native/python/pyjp_package.cpp

```diff
@@ -131,134 +136,116 @@ static PyObject *PyJPPackage_getattro(PyJPPackage *self, PyObject *attr)
 		return PyObject_GenericGetAttr((PyObject*) self, attr);
 
 	JPContext* context = JPContext_global;
-	if (context->isRunning())
+	if (!context->isRunning())
 	{
-		JPJavaFrame frame = JPJavaFrame::outer(context);
-		jobject pkg = getPackage(frame, self);
-		if (pkg == NULL)
-			return NULL;
-
-		JPPyObject out;
-		jobject obj;
-		try
-		{
-			obj = frame.getPackageObject(pkg, attrName);
-		}		catch (JPypeException& ex)
-		{
-			// If something fails, we need to go to a handler
-			if (self->m_Handler != NULL)
-			{
-				ex.toPython();
-				JPPyErrFrame err;
-				err.normalize();
-				err.clear();
-				JPPyObject tuple0 = JPPyObject::call(PyTuple_Pack(3, self, attr, err.m_ExceptionValue.get()));
-				PyObject *rc = PyObject_Call(self->m_Handler, tuple0.get(), NULL);
-				if (rc == 0)
-					return 0;
-				Py_DECREF(rc); // GCOVR_EXCL_LINE
-			}
-			throw; // GCOVR_EXCL_LINE
-		}
-		if (obj == NULL)
-		{
-			PyErr_Format(PyExc_AttributeError, "Java package '%s' has no attribute '%U'",
-					self->m_Package->m_Name.c_str(), attr);
-			return NULL;
-		} else if (frame.IsInstanceOf(obj, context->_java_lang_Class->getJavaClass()))
-			out = PyJPClass_create(frame, frame.findClass((jclass) obj));
-		else if (frame.IsInstanceOf(obj, context->_java_lang_String->getJavaClass()))
-		{
-			JPPyObject u = JPPyObject::call(PyUnicode_FromFormat("%s.%U",
-					self->m_Package->m_Name.c_str(), attr));
-			JPPyObject args = JPPyObject::call(PyTuple_Pack(1, u.get()));
-			out = JPPyObject::call(PyObject_Call((PyObject*) PyJPPackage_Type, args.get(), NULL));
-		} else
+		PyErr_Format(PyExc_RuntimeError,
+				"Unable to import '%s.%U' without JVM",
+				PyModule_GetName(self), attr);
+		return 0;
```

Comment:
Should it return a null pointer or Py_NONE? I'm not sure what happens to the callee of this function, if it retrieves a null pointer or will the set Python exception hit first?
At least we should return NULL and not mix it with plain integers, IMO.

Reply:
Quite right it should have returned NULL.   Python expects a null pointer to be returned if ever the exception is set.   It least in a few places if the exception is set Python may fail to clean up the reference.   It is considered a SystemError to return anything but NULL when returning an exception.   

I will get that bad return addressed.